### PR TITLE
merge dev inti master

### PR DESCRIPTION
**Pull request by Agipas**
Finish 6 lab

### CODE REVIEW

**Review by PetroRospopa — COMMENTED** [review]
Need fixes

File: NulpAppApi/Controller/EmptyView.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    @IBOutlet weak var view: UIView!
+    @IBOutlet weak var dataLabel: UILabel!
+    /*
+    // Only override draw() if you perform custom drawing.
+    // An empty implementation adversely affects performance during animation.
+    override func draw(_ rect: CGRect) {
+        // Drawing code
+    }
+    */
```
Comment (MarkoKuspis) [inline NulpAppApi/Controller/EmptyView.swift:21]: This should be removed

File: NulpAppApi/Controller/FavouritesTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+//  NulpAppApi
+//
+//  Created by Ivan Sapiga on 10/15/18.
+//  Copyright © 2018 Ivan Sapiga. All rights reserved.
+//
+
+import UIKit
+
+class FavouritesTableViewController: UITableViewController {
+    let placesArray = FavouriteManager.getFavourites()
```
Comment (MarkoKuspis) [inline NulpAppApi/Controller/FavouritesTableViewController.swift:12]: This is not an MVP approach. Correct way is to create presenter instance for this ViewController and then tell him when view is shown/hidden. Then presenter tells what items should be rendered

File: NulpAppApi/Controller/FavouritesTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: String(describing: FollowerTableViewCell.self), for: indexPath) as! FollowerTableViewCell
+        let follower = placesArray[indexPath.row]
+        cell.configureWith(follower: follower)
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        guard editingStyle == .delete else { return }
+        FavouriteManager.delateFavorite(id:indexPath.row)
```
Comment (MarkoKuspis) [inline NulpAppApi/Controller/FavouritesTableViewController.swift:39]: Same here. View controller should tell it's presenter to remove item, then presenter can talk to your model

**Review by MarkoKuspis — COMMENTED** [review]
Thats not an MVP-like project.
Key points:

- There is no presentation layer in your project
- View talks to model directly, which defeats the purpose of MVP
- Your `FollowerManager` and `NetworkManager` classes do too many things and should decomposed into smaller classes

**Pull request merged**